### PR TITLE
Use UTC for timestamps consistently to avoid proto merge panics

### LINKED ISSUE #5691: panic in proto merge when deep-copying server value
### Description

**What happened**:
Very often, local integrations tests fail with:
```
panic: merger not found for type:int [recovered]
	panic: merger not found for type:int

goroutine 26 [running]:
testing.tRunner.func1.1(0xecb400, 0xc0003d43f0)
	/home/deemok/dev/go/go/src/testing/testing.go:1072 +0x30d
testing.tRunner.func1(0xc000103080)
	/home/deemok/dev/go/go/src/testing/testing.go:1075 +0x41a
panic(0xecb400, 0xc0003d43f0)
	/home/deemok/dev/go/go/src/runtime/panic.go:969 +0x1b9
github.com/gogo/protobuf/proto.(*mergeInfo).computeMergeInfo(0xc0003daa80)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gogo/protobuf/proto/table_merge.go:664 +0xfff
github.com/gogo/protobuf/proto.(*mergeInfo).merge(0xc0003daa80, 0xc0002eb760, 0xc0002267e0)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gogo/protobuf/proto/table_merge.go:113 +0x4d7
github.com/gogo/protobuf/proto.(*mergeInfo).computeMergeInfo.func27(0xc0003f27f0, 0x16e56b0)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gogo/protobuf/proto/table_merge.go:545 +0x386
github.com/gogo/protobuf/proto.(*mergeInfo).merge(0xc0003daa00, 0xc0003f27e0, 0x16e56a0)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gogo/protobuf/proto/table_merge.go:139 +0x49f
github.com/gogo/protobuf/proto.(*mergeInfo).computeMergeInfo.func30(0xc0002eb650, 0xc0002eb4b0)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gogo/protobuf/proto/table_merge.go:587 +0x5c
github.com/gogo/protobuf/proto.(*mergeInfo).merge(0xc0003da980, 0xc0002eb640, 0xc0002eb4a0)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gogo/protobuf/proto/table_merge.go:139 +0x49f
github.com/gogo/protobuf/proto.(*mergeInfo).computeMergeInfo.func30(0xc0003c6a68, 0xc0003c67e8)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gogo/protobuf/proto/table_merge.go:587 +0x5c
github.com/gogo/protobuf/proto.(*mergeInfo).merge(0xc0003da8c0, 0xc0003c6a30, 0xc0003c67b0)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gogo/protobuf/proto/table_merge.go:139 +0x49f
github.com/gogo/protobuf/proto.(*mergeInfo).computeMergeInfo.func28(0xc0003c6a30, 0xc0003c67b0)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gogo/protobuf/proto/table_merge.go:555 +0x3e
github.com/gogo/protobuf/proto.(*mergeInfo).merge(0xc0003da380, 0xc0003c6a00, 0xc0003c6780)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gogo/protobuf/proto/table_merge.go:139 +0x49f
github.com/gogo/protobuf/proto.(*InternalMessageInfo).Merge(0x16e46a0, 0x1140fc0, 0xc0003c6a00, 0x1140fc0, 0xc0003c6780)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gogo/protobuf/proto/table_merge.go:50 +0x4c
github.com/gravitational/teleport/api/types.(*ServerV2).XXX_Merge(0xc0003c6a00, 0x1140fc0, 0xc0003c6780)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gravitational/teleport/api/types/types.pb.go:580 +0x57
github.com/gogo/protobuf/proto.Merge(0x1140fc0, 0xc0003c6a00, 0x1140fc0, 0xc0003c6780)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gogo/protobuf/proto/clone.go:95 +0x346
github.com/gogo/protobuf/proto.Clone(0x1140fc0, 0xc0003c6780, 0x100b5b1, 0x5)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gogo/protobuf/proto/clone.go:52 +0x19b
github.com/gravitational/teleport/api/types.(*ServerV2).DeepCopy(...)
	/home/deemok/go/src/github.com/gravitational/teleport-opt/vendor/github.com/gravitational/teleport/api/types/server.go:351
...
```
The changes I made to mitigate another server test race by patching gogo implementation of [proto.Clone](https://github.com/gravitational/teleport/pull/4790) does not seem to have the necessary effect.
The issue at hand is described [here](https://github.com/gogo/protobuf/issues/519).

I'm going to rewrite the deep copy code without `proto.Clone` as it seems to be entirely unsuited for our needs.

**What you expected to happen**:
No panic.

### Reproduction Steps
```
go test -count=1 ./integration/...
```

### PR DESCRIPTION
**Pull request by a-palchikov**
Use UTC for timestamps when unmarshaling server values to avoid issues with proto-based type merge panics.

Fixes https://github.com/gravitational/teleport/issues/5691.

### CODE REVIEW

File: lib/services/services_test.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func (s *ServicesSuite) TestLabelKeyValidation(c *check.C) {
 func TestServerDeepCopy(t *testing.T) {
 	t.Parallel()
 	// setup
-	now := time.Date(1984, time.April, 4, 0, 0, 0, 0, time.UTC)
+	now := time.Date(1984, time.April, 4, 0, 0, 0, 0, time.Local)
```
Comment (a-palchikov) [inline lib/services/services_test.go:111]: This tests the proto-based implementation deficiency (if we continued to use the implementation).

File: api/types/server.go [hunk 2]
```diff
@@ -335,7 +334,18 @@ func (s *ServerV2) CheckAndSetDefaults() error {
 
 // DeepCopy creates a clone of this server value
 func (s *ServerV2) DeepCopy() Server {
-	return proto.Clone(s).(*ServerV2)
+	newServer := *s
```
Comment (a-palchikov) [inline api/types/server.go:337]: Potential drawback is this needs attention if the server type is modified to include more pointer fields that would require cloning.

**Comment by russjones** [comment]
@r0mant @awly Do you two mind reviewing this one?

File: lib/services/servers_test.go [hunk 3]
```diff
@@ -200,7 +200,7 @@ func TestUnmarshalServerResourceKubernetes(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.desc, func(t *testing.T) {
-			got, err := UnmarshalServerResource([]byte(tt.in), KindKubeService, &MarshalConfig{})
+			got, err := UnmarshalServer([]byte(tt.in), KindKubeService)
```
Comment (awly) [inline lib/services/servers_test.go:203]: nit: update the test name accordingly

File: api/types/server.go [hunk 4]
```diff
@@ -335,7 +334,18 @@ func (s *ServerV2) CheckAndSetDefaults() error {
 
 // DeepCopy creates a clone of this server value
 func (s *ServerV2) DeepCopy() Server {
```
Comment (awly) [inline api/types/server.go:336]: A few other things to copy manually:
- `s.Spec.CmdLabels`
- `s.Spec.Apps`
- `s.Spec.KubernetesClusters`

**Review by awly — COMMENTED** [review]
It's a real shame we can't use `proto.Clone`, this manual cloning adds maintenance burden and subtle bug risks.

I think the main reason is that we use `gogoprotobuf` generator, mixed with the official `protobuf` libraries. If we could use only the official one, a lot of these problems could go away.
But `gogoproto` adds some conveniences that would be hard to give up.

**Comment by a-palchikov** [comment]
> It's a real shame we can't use `proto.Clone`, this manual cloning adds maintenance burden and subtle bug risks.
> 
> I think the main reason is that we use `gogoprotobuf` generator, mixed with the official `protobuf` libraries. If we could use only the official one, a lot of these problems could go away.
> But `gogoproto` adds some conveniences that would be hard to give up.

In fact, I may have rushed things a bit - the reason I removed the `proto.Clone` is that I was wary of other possible sources of inconsistencies with unsupported types. The "unsupported" at this point was the `time.zone` which only shows when one uses a time zone instead of UTC. If we decide we can patch all places where this might be a problem - e.g. in this case, servers were marshaled with a local timezone setting for `LastRotated` which triggered the panic after unmarshaling and subsequent `proto.Clone`. The workaround was consistently using UTC for timestamps. Even the custom `Merger` implementation aren't necessary as previously.

File: lib/services/services_test.go (hunk 1 above)
Comment (a-palchikov) [reply lib/services/services_test.go]: Reverted to use UTC again as this triggers panic.

File: api/types/server.go (hunk 2 above)
Comment (a-palchikov) [reply api/types/server.go]: Reverted to `proto.Clone` provided that timestamps are all UTC.

File: lib/services/servers_test.go (hunk 3 above)
Comment (a-palchikov) [reply lib/services/servers_test.go:203]: Updated in aec288bb10271e0943859b52192a518bd4908e98

**Review by awly — APPROVED** [review]

**Comment by russjones** [comment]
@fspmarshall Can you review?

**Review by russjones — APPROVED** [review]

**Pull request merged**